Generators/src/GeneratorPythia8.cxx

```diff
@@ -192,19 +192,41 @@ Bool_t
 
 /*****************************************************************/
 
-void GeneratorPythia8::updateHeader(FairMCEventHeader* eventHeader)
+void GeneratorPythia8::updateHeader(o2::dataformats::MCEventHeader* eventHeader)
 {
   /** update header **/
 
+  eventHeader->putInfo<std::string>("generator", "pythia8");
+  eventHeader->putInfo<int>("version", PYTHIA_VERSION_INTEGER);
```

Comment:
I think this is nice.

Do you also have an example how one can retrieve the information in a useful way?
I guess it can be in the responsibility of the concrete generator class to read back these values and offer an interface to the user to query them. (Only this generator code knows what types and keys have been put and it might be difficult for the user to read back the data directly). Alternatively, each generator could have an associated strongly modelled "Info" class/struct and we attach just one key "generatorinfo"?

Reply:
hi @sawenzel ,
I would strongly avoid to have classes that are generator dependent to store the information.
This is a bit what is in AliRoot and have caused several updates till now.
I would consider this sort of information somewhat user-dependent and this is why I suggest not to generally provide an interface for that in the core O2.
One possibility I can see to do something like what you suggest is to have it implemented in the `O2DPG` repository.

One of the clients of this mechanism is the embedding: information can be retrieved and used to generate the signal event according to what has happened in the background. I can provide an example for that use.

Another client is users, I have a simple macro that opens the kine file and reads this information.
I can put that in `Genenators/share` as an example for the time being.